common/src/main/java/io/netty/util/internal/RecyclableArrayList.java

```diff
@@ -53,6 +55,7 @@ public static RecyclableArrayList newInstance() {
     public static RecyclableArrayList newInstance(int minCapacity) {
         RecyclableArrayList ret = RECYCLER.get();
         ret.ensureCapacity(minCapacity);
+        ret.added = false;
```

Comment:
Would it make sense to move this to the `recycle` method? Should be false by default.